Extend the functionality of the i18n:remove-html-tags command

g011um opened the pull request:
The i18n:remove-html-tags command has been extended to include more
tables, thanks to a submission to the AtoM list by Clara Rosales.
https://groups.google.com/d/msg/ica-atom-users/_xdBK0ucegg/RQnNM5DKBAAJ

HTML character entities (ccedil, ecirc etc) are also translated back to
their non-HTML entity representations.

Commit:
Extend the functionality of the i18n:remove-html-tags command

The i18n:remove-html-tags command has been extended to include more
tables, thanks to a submission to the AtoM list by Clara Rosales.
https://groups.google.com/d/msg/ica-atom-users/_xdBK0ucegg/RQnNM5DKBAAJ

HTML character entities (ccedil, ecirc etc) are also translated back to
their non-HTML entity representations.

g011um commented:
This is a fairly minor change to the code really.  It changes the "public static $columns" from an array of column names to a nested array of table and columns names, then wraps the existing Artefactual code in a foreach loop.

An additional check is made to determine if the field contains HTML entities (in addition to the existing check that looked for just HTML tags).  That will cause HTML entities like **& ccedil ;** to be turned back into proper characters (like **ç**)

fiver-watson commented:
Thanks for submitting this, @g011um ! I have created the following issue ticket: 

* https://projects.artefactual.com/issues/11207

I will be asking one of our developers to code review this soon. Cheers!

MikeFE commented on lib/task/i18n/i18nRemoveHtmlTagsTask.class.php:
```diff
@@ -128,23 +179,23 @@ public function execute($arguments = array(), $options = array())
    * @return integer  number of columns changed
    */
-  private function processInformationObjectI18nHtml(&$io)
+  private function processInformationObjectI18nHtml(&$io, $tableName, $columns)
```
Can we rename this function since it no longer is specific to information objects?

g011um replied:
Do you have a suggestion?  Is processI18nHtml too generic?

MikeFE replied:
I think processI18nHtml is fine. :+1:

MikeFE commented on lib/task/i18n/i18nRemoveHtmlTagsTask.class.php:
```diff
@@ -161,15 +212,16 @@ private function processInformationObjectI18nHtml(&$io)
     {
       // Only update if tags are found
-      if ($value != strip_tags($value)) {
+      //if ($value != strip_tags($value)) {
```
Can probably just remote this line now.

g011um replied:
Sorry Mike; that's sloppy.  Didn't mean to commit that.  Easy fix.

MikeFE commented:
Looks pretty good g011um! I had 2 suggestions, other than that, there are a couple stylistic nitpicks but I noticed this task is already violating a bunch of our style rules. I'll take care of those if you implement my 2 suggestions.

Thanks!

Commit:
Removed extraneous comment

Commit:
Renamed method processInformationObjectI18nHtml to processI18nHtml

The method was no longer updating just Information Objects, so it was
renamed to be more generic. The doc block was also updated accordingly.

MikeFE commented on lib/task/i18n/i18nRemoveHtmlTagsTask.class.php:
```diff
@@ -80,32 +129,34 @@ public function execute($arguments = array(), $options = array())
-        $changedCount++;
-        $columnsChangedCount += $columnsChanged;
+        // Process HTML in row's columns
+        $columnsChanged = $this->processInformationObjectI18nHtml($io, $tableName, $columns);
```
Don't forget to rename here too :P

Commit:
Amend commit b7d55cc

Corrected call to renamed method

MikeFE commented:
Thanks for the submission g011um.

committed 9c71043cf2e8a1fdd300c5bbb9c7526efb721b86
